### PR TITLE
Add cmake-format check

### PR DESCRIPTION
**Pull request by achamayou**
We already format C++, Python and Bash consistently, so why not CMake?

### CODE REVIEW

File: .cmake-format.py [hunk 1]
```diff
@@ -0,0 +1,212 @@
+# ----------------------------------
+# Options affecting listfile parsing
+# ----------------------------------
+with section("parse"):
+
+  # Specify structure for custom cmake functions
+  additional_commands = { 'foo': { 'flags': ['BAR', 'BAZ'],
```
Comment (eddyashton) [inline .cmake-format.py]: We should specify this, at least for our custom functions that use `cmake_parse_arguments`. Something like:

```

    additional_commands = {
        "add_enclave_lib": {
            "kwargs": {"SRCS": "*", "INCLUDE_DIRS": "*", "LINK_LIBS": "*"},
        },
        "add_client_exe": {
            "kwargs": {"SRCS": "*", "INCLUDE_DIRS": "*", "LINK_LIBS": "*"},
        },
        "add_e2e_test": {
            "kwargs": {
                "NAME": "*",
                "PYTHON_SCRIPT": "*",
                "IS_SUITE": "*",
                "CURL_CLIENT": "*",
                "ADDITIONAL_ARGS": "*",
            },
        },
        "add_perf_test": {
            "kwargs": {
                "NAME": "*",
                "PYTHON_SCRIPT": "*",
                "CLIENT_BIN": "*",
                "VERIFICATION_FILE": "*",
                "LABEL": "*",
                "ADDITIONAL_ARGS": "*",
            },
        },
        "add_e2e_test": {
            "kwargs": {"SRCS": "*", "INCLUDE_DIRS": "*", "LINK_LIBS": "*",},
        },
    }
```

(though `add_enclave_lib` will be renamed by #715)

Comment (achamayou) [reply .cmake-format.py]: Nice!

**Comment by cimetrics** [comment]
cmake_format@[4211 aka 20200128.53](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4211&view=results) vs master ewma over 20 builds from [3940](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=3940&view=results) to [4190](https://msrc-ccf.visualstudio.com/CCF/_build/results?buildId=4190&view=results)
images

**Pull request merged**